### PR TITLE
Update FDA to handle groupasymmetries, multiperiod and co-added data

### LINKED ISSUE #24748: FDA 2.0 minor issues
The FDA v2.0 has some missing functionality:

1. It does not get the list of group asymmetries (currently missing in the GUI) - https://github.com/mantidproject/mantid/pull/24749/files#diff-65ed3f3882f60736d0f406757ad38ed7R87
2. I have not tested/considered it with multi period data - https://github.com/mantidproject/mantid/pull/24749/files#diff-65ed3f3882f60736d0f406757ad38ed7R98
3. Rebin/raw data should use different workspaces (assuming Rebin is not 1).  https://github.com/mantidproject/mantid/pull/24749/files#diff-d3f5b91fad30df351c8c5bd8a7989c96R125  https://github.com/mantidproject/mantid/pull/24749/files#diff-d3f5b91fad30df351c8c5bd8a7989c96R135  https://github.com/mantidproject/mantid/pull/24749/files#diff-d3f5b91fad30df351c8c5bd8a7989c96R154

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This updates the Frequency Domain Analysis 2 GUI to handle group asymmetries, multiperiod data and co-added data correctly. 

**Report to:** anthony. 

**To test:**

 * Open the new GUI and load some multiperiod data i.e.(HIFI84447). 
 * Check that the correct options appear in the Transform widget.
 * Change the periods to use on the home tab, click update all (this should happen automatically after #24835 ) and check again that the options are correct.
  * Try to Co-Add some data and check that the correct options appear in the transform tab

Partially Fixes #24748  

*This does not require release notes* because This GUI is not yet user facing

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py [hunk 1]
```diff
@@ -127,7 +126,7 @@ def handle_load_grouping_from_file(self):
         self.pairing_table_widget.update_view_from_model()
         self.update_description_text(description)
 
-        self.groupingNotifier.notify_subscribers()
+        # self.groupingNotifier.notify_subscribers()
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py]: why is this commented out?

File: scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py [hunk 2]
```diff
@@ -145,25 +144,29 @@ def calculate_all_data(self):
...
+        self.groupingNotifier.notify_subscribers()
+
     def handle_default_grouping_button_clicked(self):
         self._model.reset_groups_and_pairs_to_default()
         self.grouping_table_widget.update_view_from_model()
         self.pairing_table_widget.update_view_from_model()
         self.update_description_text()
 
-        self.groupingNotifier.notify_subscribers()
+        # self.groupingNotifier.notify_subscribers()
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py]: comment

File: scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py [hunk 3]
```diff
@@ -145,25 +144,29 @@ def calculate_all_data(self):
...
 
     def on_clear_requested(self):
         self._model.clear()
         self.grouping_table_widget.update_view_from_model()
         self.pairing_table_widget.update_view_from_model()
         self.update_description_text()
 
-        self.groupingNotifier.notify_subscribers()
+        # self.groupingNotifier.notify_subscribers()
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py]: comment

**Comment by AnthonyLim23** [comment]
I broke it. If you load HIFI 84447 and try and do a maxent or a phaseQuad it falls over as it does not know which period to use

**Comment by Mantid-Matthew** [comment]
I have fixed the issue with phaseQuad and maxent with multiperiods in #24876

File: scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py [hunk 4]
```diff
@@ -186,10 +183,12 @@ def create_update_thread(self):
...
 
     def group_table_changed(self):
-        self.groupingNotifier.notify_subscribers()
+        pass
+        # self.groupingNotifier.notify_subscribers()
 
     def pair_table_changed(self):
-        self.groupingNotifier.notify_subscribers()
+        pass
+        # self.groupingNotifier.notify_subscribers()
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py]: two more here

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please?

**Pull request merged**